portable/ThirdParty/GCC/RP2040/library.cmake

```diff
@@ -0,0 +1,67 @@
+# Called after the Raspberry Pi Pico SDK has been initialized to add our libraries
+
+add_library(FreeRTOS-Kernel_included INTERFACE)
+target_compile_definitions(FreeRTOS-Kernel_included INTERFACE
+        -DFREERTOS_KERNEL=1
+        # todo don't know if we need this; maybe just want to explicitly disable things
+        -DPICO_RTOS=1 # let SDK know that RTOS is in use
+        )
+
+pico_add_platform_library(FreeRTOS-Kernel_included)
+
+add_library(FreeRTOS-Kernel-Core INTERFACE)
+target_sources(FreeRTOS-Kernel-Core INTERFACE
+        ${FREERTOS_KERNEL_PATH}/croutine.c
+        ${FREERTOS_KERNEL_PATH}/event_groups.c
+        ${FREERTOS_KERNEL_PATH}/list.c
+        ${FREERTOS_KERNEL_PATH}/queue.c
+        ${FREERTOS_KERNEL_PATH}/stream_buffer.c
+        ${FREERTOS_KERNEL_PATH}/tasks.c
+        ${FREERTOS_KERNEL_PATH}/timers.c
+        )
+target_include_directories(FreeRTOS-Kernel-Core INTERFACE ${FREERTOS_KERNEL_PATH}/include)
+
+add_library(FreeRTOS-Kernel INTERFACE)
+target_sources(FreeRTOS-Kernel INTERFACE
+        ${CMAKE_CURRENT_LIST_DIR}/port.c
+)
+
+target_include_directories(FreeRTOS-Kernel INTERFACE
+        ${CMAKE_CURRENT_LIST_DIR}/include)
+
+target_link_libraries(FreeRTOS-Kernel INTERFACE
+        FreeRTOS-Kernel-Core
+        pico_base_headers
+        hardware_exception)
+
+target_compile_definitions(FreeRTOS-Kernel INTERFACE
```

Comment:
Could you comment on your choice of INTERFACE rather than STATIC for the library definition?

Reply:
All libraries within the Raspberry Pi Pico SDK build as INTERFACE libraries to gather build options/configuration settings on a per target binary basis. i.e. there may be multiple target executables in a single cmake build that use differently configured FreeRTOS and or SDK settings